### PR TITLE
solving doc test failures

### PR DESCRIPTION
**Pull request by hantek**
Solves test failures in the doc tests. Fix for #3103 .

### CODE REVIEW

File: doc/tutorial/profiling.txt [hunk 1]
```diff
@@ -33,7 +33,7 @@ functions using either of the following two options:
     - You can also combine the profile of many functions: 
     
       .. testcode::
-
+          :hide:
```
Comment (nouiz) [inline doc/tutorial/profiling.txt]: This cause error in travis. I think you need an empty line after hide.

File: doc/tutorial/profiling.txt [hunk 2]
```diff
@@ -33,11 +33,12 @@ functions using either of the following two options:
...
       .. testcode::
+          :hide:
 
           profile = theano.compile.ProfileStats()
-          f = theano.function(..., profile=profile)
-          g = theano.function(..., profile=profile)
-          ...
+          f = theano.function(..., profile=profile)  # doctest: +SKIP
+          g = theano.function(..., profile=profile)  # doctest: +SKIP
+          ...  # doctest: +SKIP
```
Comment (hantek) [inline doc/tutorial/profiling.txt]: Adding a blank line still doesn't solve the problem. What we need here is to have a testcode block that doesn't actually get executed. I also tried adding "# doctest: +skip" at the end of each line, but it didn't work too.

Comment (hantek) [inline doc/tutorial/profiling.txt]: I changed the block type to "doctest" instead, which made "# doctest: + SKIP" work. The layout of "doctest" in html seems to be the same to "testcode."

**Comment by nouiz** [comment]
You need to modify the .travis.yml file to uncomment the execution of tests. line 69 in master.

**Comment by hantek** [comment]
Hi Fred, Travis has passed, it looks ready for merge now.

**Comment by hantek** [comment]
There is some problem in python 3.3, like it can't import cPickle, can't recognize "file," etc.  What did we  usually do on these?

**Comment by nouiz** [comment]
In python 3, xrange don't exist. Just use range instead.

In Python 3; "print 3" don't exist. You should use "print(3)" instead. print is now a function.

Fix the import of cPickle by using this import "from six.moves import cPickle"

For the function file, I think you can replace it with open()

**Comment by hantek** [comment]
Yes but what confused me is that, why these test got passed in python3.3 previously but starts to fail from the last commit? Nothing relating to that was changed.

**Comment by nouiz** [comment]
In the commit  b5d75ef, you added the test back in travis, but travis didn't ran for this commit. I don't know why, this is why it got marked as passed (travis didn't run, so it didn't fail).

**Comment by hantek** [comment]
Hi Fred, I updated the last commit correspondingly.

**Comment by nouiz** [comment]
We can merge when travis pass.

**Comment by nouiz** [comment]
thanks

**Pull request merged**